src/main/java/kr/hhplus/be/server/application/order/OrderFacade.java

```diff
@@ -0,0 +1,129 @@
+import kr.hhplus.be.server.application.coupon.CouponService;
+import kr.hhplus.be.server.application.coupon.UseCouponCommand;
+import kr.hhplus.be.server.application.coupon.UseCouponInfo;
+import kr.hhplus.be.server.application.order.service.OrderItemService;
+import kr.hhplus.be.server.application.order.service.OrderService;
+import kr.hhplus.be.server.application.payment.PaymentService;
+import kr.hhplus.be.server.application.payment.RequestPaymentCommand;
+import kr.hhplus.be.server.application.point.PointService;
+import kr.hhplus.be.server.application.point.UpdatePointCommand;
+import kr.hhplus.be.server.application.product.ProductService;
+import kr.hhplus.be.server.application.user.UserService;
+import kr.hhplus.be.server.domain.coupon.Coupon;
+import kr.hhplus.be.server.domain.order.Order;
+import kr.hhplus.be.server.domain.order.OrderItem;
+import kr.hhplus.be.server.domain.payment.Payment;
+import kr.hhplus.be.server.domain.product.Product;
+import kr.hhplus.be.server.domain.user.User;
+import kr.hhplus.be.server.support.common.exception.CustomException;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+@RequiredArgsConstructor
+public class OrderFacade {
+
+	private final Logger logger = LoggerFactory.getLogger(OrderFacade.class);
+	private final OrderService orderService;
+	private final PaymentService paymentService;
+	private final ProductService productService;
+	private final PointService pointService;
+	private final CouponService couponService;
+	private final OrderItemService orderItemService;
+	private final UserService userService;
+
+	public OrderInfoResult getOrders(Long userId) {
+		logger.info("### getOrders parameter : {}", userId);
+		List<OrderDetailResult> orderDetails = orderService.findHistoryByUserId(userId)
+			.stream().map(oh -> {
+				Long orderId = oh.getOrder().getId();
+				List<Product> products = productService.orderItemToProduct(
+					oh.getOrder().getOrderItems());
+				List<Coupon> coupons = oh.getOrder().getCouponIds().stream()
+					.map(couponService::findCouponById)
+					.toList();
+				return new OrderDetailResult(orderId, products, coupons);
+			}).toList();
+		return OrderInfoResult.from(userId, orderDetails);
+	}
+
+	public CancelOrderResult cancel(CancelOrderCriteria criteria) {
+		logger.info("### cancel parameter : {}", criteria.toString());
+		Order order = orderService.findByIdAndUserId(criteria.orderId(), criteria.userId());
+		User user = userService.get(criteria.userId());
+		// 결제 취소
+		Payment payment = paymentService.cancel(order);
+		// 포인트 환불
+		pointService.refund(UpdatePointCommand.Refund.of(user.getId(), order.getTotalPrice()));
+		// 쿠폰 상태 복원 (쿠폰 사용했으면)
+		if (!order.getCouponIds().isEmpty()) {
+			couponService.restore(UseCouponCommand.of(user.getId(), order.getCouponIds()));
+		}
```

Comment:
```suggestion
		if (order.isUsedCoupon()) {
			couponService.restore(UseCouponCommand.of(user.getId(), order.getCouponIds()));
		}
```

`!order.getCouponIds().isEmpty()` 보다 `Order`클래스에 "메세지"를 보내면 가독성 향상에 좋아질 것 같습니다!